src/shims/foreign_items.rs

```diff
@@ -465,26 +466,8 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriEvalContextExt<'mir, 'tcx
                     }
```

Comment:
I think basically the entire body pf `setenv` should move into `envs.rs`, so all that is left here is something like
```rust
                    let name_ptr = this.read_scalar(args[0])?.not_undef()?;
                    let value_ptr = this.read_scalar(args[1])?.not_undef()?;
                    EnvVars::setenv(this, name_ptr, value_ptr)?;
```

Reply:
> `EnvVars::setenv(this, name_ptr, value_ptr)?;`

Actually we can do better, copy what we do [here](https://github.com/rust-lang/miri/blob/master/src/shims/tls.rs#L141) for TLS.